Base/Python/slicer/parameterNodeWrapper/wrapper.py

```diff
@@ -86,9 +87,17 @@ class _CachedParameterWrapper(_ParameterWrapper):
     def __init__(self, parameter: _Parameter, parameterNode):
         super().__init__(parameter, parameterNode)
         self._value = self.parameter.read(self.parameterNode)
-        self._observerTag: int = parameterNode.AddObserver(vtk.vtkCommand.ModifiedEvent, self._onModified)
+        # Important: We don't want to increase the reference to self here by including it in the AddObserver callback.
+        # This would prevent the object from being garbage collected, and the observers from being removed when the object goes out of scope.
+        # Instead, we create a weakref to self and use it in a lambda function.
+        _selfWeakRef = weakref.ref(self)
+        self._observerTag: int = parameterNode.AddObserver(vtk.vtkCommand.ModifiedEvent,
+                                                           lambda caller, event: _selfWeakRef()._onModified(caller, event))
```

Comment:
If `self` is deleted then the lambda function will be resolved to `None._onModified(caller, event)`, which is not very nice and may have side effects. We just don't see any errors because there is no error reporting in callbacks.

What do you think about this instead?

`lambda caller, event: _CachedParameterWrapper._onModified(_selfWeakRef(), caller, event))`

Then we would start `_onModified` implementation with `if self is None: return` (with a comment there explaining why it can happen).

Reply:
Nevermind, `__del__` removes the observer, so it is not possible that an observation comes in after `self` is deleted, so `None._onModified(...)` cannot happen. So, the current implementation should be good.